### ISSUE TITLE
Question: ColumnName Resolver

### ISSUE THREAD

**Issue by andreas-soroko** [mikependon/RepoDB#387]
Hey,

in #366 i already asked if it would be possible to add a columName resolver and you gave me same examples... but i dont get it. 

**so first, some code:** 

```csharp
[Map("enum_test_table")]
private class TestModel
{
    [Primary]
    public Guid Id { get; set; }
    public JObject JsonData { get; set; }
}
```

```sql
create table enum_test_table(id uuid PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(), json_data jsonb);
```

now i thought, i need to replace this boy [PostgreSqlConvertFieldResolver](https://github.com/mikependon/RepoDb/blob/master/RepoDb.PostgreSql/RepoDb.PostgreSql/Resolvers/PostgreSqlConvertFieldResolver.cs)

and the only way i found to pass the resolver is

```csharp
StatementBuilderMapper.Add(typeof(NpgsqlConnection), GetStatementBuilder(), true);

IStatementBuilder GetStatementBuilder()
{
    return new PostgreSqlStatementBuilder(
        DbSettingMapper.Get(typeof(NpgsqlConnection)),
        new PostgresFieldResolver(), // my resolver, just 1:1 copy from yours currently, but debug points does not stop there
        new ClientTypeToAverageableClientTypeResolver());
}
```

```csharp
var model = new TestModel()
{
    Id = Guid.NewGuid(),
    JsonData = JObject.FromObject(new
    {
        testField = "data"
    })
};
await repo.InsertAsync<Guid>(model);

error -> Npgsql.PostgresException : 42703: column "Id" of relation "enum_test_table" does not exist
```

**what is my goal ?** 
- use on all props snake_case without using MapAttribute 
- If MapAttribute is defined, then it wins over the fieldResolver 

thank you for your help!

**Comment by mikependon** [comment]
As I understand it, you would like to override the column name that is being generated by the Statement Builder. There is no really field name converter concept in RepoDb. The resolver I specified is only to handle the cross-DB providers CONVERT functionality, but not the mapping.

You would like this converter to execute every time the Statement Builder tried to compose a SQL Statement for any operations -- this is not supported in RepoDb. Well, this is quite advance topic.

The alternative solution is you create your own PostgreSqlStatementBuilder and inject it - but I would suggest you should not do this as it will take time for you.

Otherwise, if this is really important, then I can add this as an additional feature.

**Comment by andreas-soroko** [comment]
>  The alternative solution is you create your own PostgreSqlStatementBuilder and inject it - but I would suggest you should not do this as it will take time for you.
> Otherwise, if this is really important, then I can add this as an additional feature.

we would wait until this feature is done

**Comment by lauw** [comment]
I just made a gist to support snake_case as that's what I'm using as well.
https://gist.github.com/lauw/20747e47e373444432278985e8eee495
 It's required that you use the latest beta version which has support for Class/PropertyMapping.

You have to add [Table] attributes to your table object classes for the mapper to find them, but edit/use as you see fit.

Add the gist to your project and call it:
`RepoDbMapper.MapTablesAndColumns(NameCasing.SnakeCase);
`

**Comment by mikependon** [comment]
@ffandreassoroko - would you be able to check if the feature implicit mapping would address your issue here? We introduced this feature so you will be able to control the mappings by yourselves through class-mappings (and not via attributes). You can see the feature [here](http://repodb.net/feature/implicitmapping).

**To map a class.**

```csharp
ClassMapper.Add<TestModel>("enum_test_table");
```

**To map a property.**

```csharp
PropertyMapper.Add<TestModel>(e => e.Property, "property_name");
```

It also supports the following (on top of the 2 mappers mentioned above).

- Identity Mapping
- Primary Mapping
- Type Mapping
- Property Handler Mapping

**Comment by mikependon** [comment]
Alternatively, we decided to use the mapper instead of resolver. This request of yours can be addressed by `FluentMapper` class as part of the latest release.

```csharp
FluentMapper
	.Entity<Customer>() // Which Entity Model?
	.Table("[sales].[Customer]") // Table/View Mapping
	.Primary(e => e.Id) // Primary Mapping
	.Identity(e => e.Id) // Identity Mapping
	.Column(e => e.Id, "CustomerId") // Column Mapping
	.Column(e => e.Name, "CustomerName") // Column Mapping
	.DbType(e => e.DateOfBirth, DbType.DateTime2) // DbType Mapping
	.PropertyHandler<CustomerAddressPropertyHandler>(e => e.Address); // PropertyHandler Mapping
```

The feature is available at [RepoDb (v1.11.0)](https://www.nuget.org/packages/RepoDb/1.11.0).

```csharp
> Install-Package RepoDb -version 1.11.0
```

Please visit the [Implicit Mapping](http://repodb.net/feature/implicitmapping) to learn more.

**Comment by mikependon** [comment]
@ffandreassoroko @lauw - hey gents! There is an interesting use-case in PostgreSQL that you may be interested. The issue can be found at #467.

To summarize it, it is all about resolving the casing of the classes and column names for PostgreSQL execution. As a response, we made a fix to support the empty `OpeningQuote` and `ClosingQuote` in the `IDbSetting`.

Alternatively, you can also use the [gist](https://gist.github.com/mikependon/a969bf819b017cf8e47c12d06d58ee3c) that I had provided for this issue named `EntityMappingLowerizer`.

**Via Entity Level:**

```csharp
EntityMappingLowerizer.ForEntity<Customer>();
var tableName = ClassMapper.Get<Customer>(); // Test the model Customer
var columnName = PropertyMapper.Get<Customer>(e => e.Name); // Test the model Customer.Name
```

**Via Namespace (All models on the namespace):**

```csharp
EntityMappingLowerizer.ForNamespace(typeof(Customer).Namespace); // All models will be affected
var tableName = ClassMapper.Get<Customer>(); // Test the model Customer
var columnName = PropertyMapper.Get<Customer>(e => e.Name); // Test the model Customer.Name
tableName = ClassMapper.Get<Person>(); // Test the model Person
```

Please do let me know if this is something that could help your cases as well. Thanks!
